### DISCUSSION TITLE
Auto Z-Offset?

### DISCUSSION THREAD

**Discussion by Avpman2** [xblax/flashforge_ad5m_klipper_mod#184] [Q&A] [answered]
Is there an automatic Z-Offest (macro or command line) with this mod? Either via Guppyscreen, Mainsail or Fluidd? All I found were commands/macros to save the offset.

**Comment by xblax** [comment]
Klipper has the "PROBE_CALIBRATE" command which assists you in doing the probe z-offset calibration. Or do you mean something different?
See https://www.klipper3d.org/Probe_Calibrate.html

**Reply by Avpman2** [reply]
Thank you, I am there now. 
Where (and which) .cfg file is the Z-Offset stored? I want to double check after calibrating.
AWESOME mod, btw!

**Reply by xblax** [reply]
It's in the printer.cfg - it has all the calibration or saved config data.

**Comment by Avpman2** [comment]
I saw that file in two places below /mnt and /root

Both have the value, what appears to be commented out? However, I admit I don’t know what the comment delims are in the config file. I could be entirely wrong?

 

#*# [probe]

#*# z_offset = -0.480

#*#

#*# [bed_mesh default .04 -0.025 zax]

#*# version = 1

#*# points =

**Reply by consp** [reply]
> Both have the value, what appears to be commented out? However, I admit I don’t know what the comment delims are in the config file. I could be entirely wrong?

Klipper uses #*# as saved overrides. These are the values you saved from the ui.

[It should be marked with this](https://github.com/Klipper3d/klipper/blob/dae8b8cacff583b180c650d9a6d1601907a12f1e/klippy/configfile.py#L135C1-L139C4):
```
AUTOSAVE_HEADER = """
#*# <---------------------- SAVE_CONFIG ---------------------->
#*# DO NOT EDIT THIS BLOCK OR BELOW. The contents are auto-generated.
#*#
"""
```

**Comment by Avpman2** [comment]
Where is the Z-Offset on the ADM5-Pro when using the stock FF FW? I'm having trouble setting the Z-Offset when using the Klipper Mod. I followed the docs but prints still don't stick. However, with the same model and the same temperature settings, I can print fine with Orca-Flashforge and the stock FW. When using Klipper the nozzle appears to drag the skirt into the model.
If I can get the Z-Offset from the stock FW, I can transfer it to the clipper config files. Otherwise, I'm dead in the water with the mod.

**Reply by consp** [reply]
Should be in /opt (on the original filesystem, so /mnt/orig_root/opt in the mod) somewhere, but FF does magic in their program and changes stuff on the fly so don't count on it being accurate.

**Reply by xblax** [reply]
The pre-configured probe z-offset by the mod is -0.25. Try changing it to -0.15 manually, that could give you better bed adhesion. Because it's not mentioned here: you have to print with bed heating on the PEI plates. Try setting it to at least 60° for PLA if you don't have it enabled already.

**Comment by Avpman2** [comment, marked as answer]
I had been using the V.05 Guppyscreen version. I just reloaded the full Klipper screen variety and changed the Z offset (in Orca) to -.15, and it works great! Dunno if it was the prior FW or what, but I'm good now. I cannot express my appreciation adequately, not only for all the hard work on the mod but also for the willingness of people on here to help. I started out a NOOB, but I'm working my way quickly to becoming a more advanced user, much of which is due to the help of everyone on here.
